apps/visualization/views.py

```diff
@@ -45,3 +57,62 @@ def get_queryset(self):
             ) for value in result
         ])
         return DataCountryLevelMostRecent.objects.filter(filters).distinct('subvariable')
+
+
+class ExportBaseView(ListAPIView):
+    pagination_class = LimitOffsetPagination
+
+    def get_serializer_class(self):
+        return None
+
+    def validate_export_params_together(self, params):
+        query_params_keys_list = list(params.keys())
+        if not any(elem in ['iso3', 'indicator_id'] for elem in query_params_keys_list):
+            raise ValidationError({'error': 'iso3 or indicator_id params are required'})
+        if 'limit' not in query_params_keys_list:
+            raise ValidationError({'error': 'Limit is required.'})
+        if 'offset' not in query_params_keys_list:
+            raise ValidationError({'error': 'offset is required.'})
+        limit = int(params.get('limit'))
+        if limit > settings.OPEN_API_MAX_EXPORT_PAGE_LIMIT:
+            raise ValidationError(
+                {'error': f'Limit must be less or equal to {settings.OPEN_API_MAX_PAGE_LIMIT}'}
+            )
```

Comment:
Also,
- We might need `non_field_errors` here instead of errors
- Send all the validation errors at once.

```suggestion
        errors = []
        if not any(elem in ['iso3', 'indicator_id'] for elem in query_params_keys_list):
            errors.append('iso3 or indicator_id params are required')
        if 'limit' not in query_params_keys_list:
            errors.append('limit is required.')
        if 'offset' not in query_params_keys_list:
            errors.append('offset is required.')
        limit = int(params.get('limit'))
        if limit > settings.OPEN_API_MAX_EXPORT_PAGE_LIMIT:
            errors.append(f'limit must be less or equal to {settings.OPEN_API_MAX_PAGE_LIMIT}')
        if errors:
            raise ValidationError({'non_field_errors': errors})
```

@AdityaKhatri This is correct right?

Reply:
@thenav56  @AdityaKhatri  Changed this one